active/AnimatedIcon/AnimatedIconSpec.md

```diff
@@ -0,0 +1,894 @@
+        <TextBlock Text='Click' />
+    </StackPanel>
+</Button>
+```
+
+Some controls will automatically set the State on an AnimatedIcon, for example
+see NavigationViewItem.Icon.
+
+## Other AnimatedIcon class members
+
+| Name | Description | Default |
+| :---| :---| :---|
+| Source | Animation data, for example generated by the LottieGen tool. | null |
+| FallbackSource | Gets or sets the static icon that will be used as a fallback when the OS cannot run the animated icon file.   | null |
+| State | Property that the developer sets on AnimatedIcon. See below for more details. | null |
+
+## IconSource.CreateIconElement method
+
+_Spec note: This is being added to make it easier to create an AnimatedIcon from an AnimatedIconSource_
+
+```cs
+public IconElement CreateIconElement();
+```
+
+Call this method to get an IconElement from an IconSource.
+
+An IconElement can be used by controls that have an IconElement property,
+or since it's an element it can be used anywhere in the XAML element tree.
+
+This example creates a BitmapIcon from a BitmapIconSource:
+
+```cs
+IconSource iconSource = new BitmapIconSource();
+BitmapIcon = iconSource.CreateIconElement as BitmapIcon;
+```
+
+## IconSource.CreateIconElementCore
+
+```cs
+protected virtual IconElement CreateIconElementCore();
+```
+
+Override this method in a subclass of IconSource to produce an IconElement
+that corresponds to your IconSource.
+
+This protected virtual CreateIconElementCore method is called
+when the public CreateIconElement method is called.
+
+_Spec note: this naming pattern of a public Foo and protected virtual FooCore is an
+existing pattern in Xaml today. The other name suffix that's used is "Override", but
+"Core" is more frequent._
+
+## NavigationViewItem.Icon
+
+_Spec note: this isn't a new property, but has new behavior_
+
+If you set an AnimatedIcon as the value of the Icon property, 
+the NavigationViewItem will set the states of the AnimatedIcon for you, according to the states of the control.
+
+For example, this sets a custom animation `PlayIcon` that was generated by the LottieGen tool:
+
+```xml
+<NavigationView.MenuItems>
+    <NavigationViewItem Content = "Play">
+        <NavigationViewItem.Icon>
+
+            <AnimatedIcon >
+                <local:PlayIcon />
+            </AnimatedIcon>
+
+    </NavigationViewItem.Icon>
+</NavigationView.MenuItems>
+```
+
+The NavigationViewItem will automatically set the following states on the AnimatedIcon:
+* Normal
+* PointerOver
+* Pressed
+* Selected
+* PressedSelected
+* PointerOverSelected
+
+If the PlayIcon has a marker for "NormalToPointerOver", it will be animated
+to when the mouse moves to over the NavigationViewItem.
+See [link] for more information on combining marker names.
+
+To create this custom PlayIcon class, you run a Lottie JSON file (with markers) through the Windows
+[CodeGen](https://docs.microsoft.com/en-us/windows/communitytoolkit/animations/lottie-scenarios/getting_started_codegen)
```

Comment:
Yes it is the same tool and that should actually say LottieGen instead of CodeGen. LottieGen is the tool the developer uses and CodeGen is the output generated from the tool.

Reply:
Recommend: pick one name and stick with it